File: control-flow/Q113/main.go
package main

import "fmt"

func main() {
	var (
		a int
		b int
		c int
	)
	fmt.Print("Enter a: ")
	fmt.Scan(&a)
	fmt.Print("Enter b: ")
	fmt.Scan(&b)
	fmt.Print("Enter c: ")
	fmt.Scan(&c)
	if a*a == b*b + c*c || b*b == a*a + c*c || c*c == a*a + b*b {
		fmt.Println("YES")
	} else {
		fmt.Println("NO")
	}
}

File: control-flow/Q113/main.rs
use std::io;

fn read_int(prompt: &str) -> i64 {
    let mut input = String::new();
    println!("{}", prompt);
    io::stdin().read_line(&mut input).expect("Failed to read line");
    input.trim().parse().expect("Not a valid number")
}

fn main() {
    let a = read_int("Enter a:");
    let b = read_int("Enter b:");
    let c = read_int("Enter c:");

    let aa = a * a;
    let bb = b * b;
    let cc = c * c;

    if aa == bb + cc || bb == aa + cc || cc == aa + bb {
        println!("YES");
    } else {
        println!("NO");
    }
}